src/cmdline/mod.rs

```diff
@@ -336,172 +250,641 @@ impl Cmdline {
     }
 }
 
+impl UpdateArgs for BootArgs {
+    fn has_capacity(&self, more: usize) -> bool {
```

Comment:
Why is this implemented for each type of Arg? We should be able to use the same implementation twice. For example, we could implement the trait for String, and then here just call `self.0.has_capacity()`. Same for InitArgs. This also applies to the other functions that are duplicated.

Reply:
yes, you are right.... still not get used to rust .... I initially tried to move has_capacity()'s implementation into impl UpdateArgs block (similar to insert_key_value()) but it was actually impossible because the logic of this method needs reference to self object of the struct for which we need the trait to be implemented and then I just find as a solution to duplicate the code... Thank you for your advice! I moved the implementation into impl UpdateArgs for String and then just called the methods from there.